### PR TITLE
V0.2 vis

### PR DESCRIPTION
**Pull request by micli**
# Description

Code implemented dump data before reset every time in Env class. With this feature, developer can generate visualization to view data trend. 
To enable data dump, please add an option: "enable-dump-snapshot" as parameters when initialize Env object.

## Linked issue(s)/Pull request(s)

- [issue_number](issue_link)

## Type of Change

- [ ] Non-breaking bug fix
- [ ] Breaking bug fix
- [x] New feature
- [ ] Test
- [ ] Doc update
- [ ] Docker update

## Related Component

- [x] Simulation toolkit
- [ ] RL toolkit
- [ ] Distributed toolkit

## Has Been Tested

- OS:
  - [x] Windows
  - [x] Mac OS
  - [x] Linux
- Python version:
  - [x] 3.6
  - [] 3.7
- Key information snapshot(s):
  
## Needs Follow Up Actions

- [x] New release package
- [ ] New docker image

## Checklist

- [x] Add/update the related comments
- [ ] Add/update the related tests
- [ ] Add/update the related documentations
- [ ] Update the dependent downstream modules usage

### CODE REVIEW

File: examples/hello_world/citi_bike/hello.py [hunk 1]
```diff
@@ -9,8 +9,15 @@
 durations = 100
 max_ep = 2
 
+opts = dict()
+""" enable-dump-snapshot parameter means business_engine needs dump snapshot data before reset.
+If you leave value to empty string, it will dump to current folder.
+For getting dump data, please uncomment below line and specify dump destination folder.
+"""
+opts['enable-dump-snapshot'] = '/Users/micl/Documents/dumps'
```
Comment (chaosddp) [inline examples/hello_world/citi_bike/hello.py:17]: remove hard coded path

File: examples/hello_world/cim/hello.py [hunk 2]
```diff
@@ -7,9 +7,12 @@
 start_tick = 0
 durations = 100  # 100 days
 
+opts = dict()
+opts['enable-dump-snapshot'] = '/Users/micl/Documents/dumps'
```
Comment (chaosddp) [inline examples/hello_world/cim/hello.py:11]: hard coded path

File: maro/data_lib/dump_csv_converter.py [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+from datetime import datetime
+from pathlib import Path
+import threading
+import pandas as pd
+import numpy as np
+from math import floor
+
+
+
+class dump_csv_converter:
```
Comment (chaosddp) [inline maro/data_lib/dump_csv_converter.py:15]: Class name should be CamelStyle

File: maro/data_lib/dump_csv_converter.py [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+                    dataframe = pd.DataFrame(csv_data)
+                    dataframe.to_csv(os.path.join(curDir, file.replace('.meta', '.csv')), index = False)
+
+    def start_processing(self):
+        thread = threading.Thread(target = self.process_data)
+        thread.start()
+
+    def get_column_info(self, filename):
+        with open(filename, 'r') as f:
+            columns = f.readline().replace('\n', '').replace('\r\n', '')
```
Comment (chaosddp) [inline maro/data_lib/dump_csv_converter.py:76]: str.strip?

File: maro/data_lib/dump_csv_converter.py [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+                if file.endswith('.meta') or file.endswith('.npy'):
+                    os.remove(file)
+
+    def dump_descsion_events(self, decision_events, start_tick: int, resolution: int):
+        decision_events_file = os.path.join(self._foldername, 'decision_events.csv')
+        headers, colums_count = self._calc_event_headers(decision_events[0])
+        array = []
+        for event in decision_events:
+            attr_dict = event.__getstate__()
+            if attr_dict.__contains__('tick'):
```
Comment (chaosddp) [inline maro/data_lib/dump_csv_converter.py:100]: "key" in dict style would be better and clean

File: maro/data_lib/dump_csv_converter.py [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+        headers = []
+        count = 0
+        for attr in dir(event):
+            if attr[0] != '_':
+                headers.append(attr)
+                count = count + 1
+
+        return headers, count
+
+    def tick_to_frame_index(self, start_tick: int, cur_tick: int, resolution: int) -> int:
```
Comment (chaosddp) [inline maro/data_lib/dump_csv_converter.py:123]: dup code from simulator.utils.common?

**Comment by ArthurJiang** [comment]
@micli  please fix the lint error.

File: maro/simulator/core.py [hunk 7]
```diff
@@ -92,6 +96,17 @@ def reset(self):
 
         self._event_buffer.reset()
 
+        if self._additional_options.__contains__('enable-dump-snapshot'):
```
[the remark this answers is not in the document]
Comment (micli) [reply maro/simulator/core.py]: Already changed to if 'enable-dump-snapshot' in self._additional_options:

File: examples/cim/dqn/single_process_launcher.py [hunk 8]
```diff
@@ -16,10 +16,16 @@
 from components.experience_shaper import TruncatedExperienceShaper
 from components.state_shaper import CIMStateShaper
 
+opts = dict()
```
[the remark this answers is not in the document]
Comment (micli) [reply examples/cim/dqn/single_process_launcher.py]: Already removed.

File: examples/hello_world/citi_bike/hello.py (hunk 1 above)
Comment (micli) [reply examples/hello_world/citi_bike/hello.py]: Already removed.

File: examples/hello_world/cim/hello.py (hunk 2 above)
Comment (micli) [reply examples/hello_world/cim/hello.py]: Already removed.

File: maro/data_lib/dump_csv_converter.py (hunk 3 above)
Comment (micli) [reply maro/data_lib/dump_csv_converter.py]: Already renamed.

File: maro/data_lib/dump_csv_converter.py (hunk 4 above)
Comment (micli) [reply maro/data_lib/dump_csv_converter.py]: Already replaced with str.strip()

File: maro/data_lib/dump_csv_converter.py (hunk 6 above)
Comment (micli) [reply maro/data_lib/dump_csv_converter.py]: Already removed.

**Pull request merged**